aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client.md

```diff
@@ -29,106 +29,141 @@ In this tutorial, a client app is written that consumes the following web API.
 | Update a product | PUT | /api/products/*id* |
 | Delete a product | DELETE | /api/products/*id* |
 
-> [!NOTE]
-> To learn how to implement this API on the server, using ASP.NET Web API, see [Creating a Web API that Supports CRUD Operations](../older-versions/creating-a-web-api-that-supports-crud-operations.md).
-
+To learn how to implement this API with ASP.NET Web API, see [Creating a Web API that Supports CRUD Operations](xref:web-api/overview/getting-started-with-aspnet-web-api/tutorial-your-first-web-api
+).
 
 For simplicity, the client application in this tutorial is a Windows console application. **HttpClient** is also supported for Windows Phone and Windows Store apps. For more information, see [Writing Web API Client Code for Multiple Platforms Using Portable Libraries](https://blogs.msdn.com/b/webdev/archive/2013/07/19/writing-web-api-client-code-for-multiple-platforms-using-portable-libraries.aspx)
 
 <a id="CreateConsoleApp"></a>
 ## Create the Console Application
 
-In Visual Studio, create a new Windows console application and paste in the following code.
+In Visual Studio, create a new Windows console app named **HttpClientSample** and paste in the following code:
+
+[!code-csharp[Main](calling-a-web-api-from-a-net-client/sample/client/Program.cs?name=snippet_all)]
+
+The preceding code is the complete client app.
 
-[!code-csharp[Main](calling-a-web-api-from-a-net-client/samples/sample1.cs)]
+`RunAsync` runs and blocks until it completes. Most **HttpClient** methods are async, because they perform network I/O. All of the async tasks are done inside `RunAsync`. In a console application, it's OK to block the main thread inside of `Main`. In a GUI application, you should **not** block the UI thread.
```

Comment:
"In a console application, it's OK to block the main thread inside of `Main`. In a GUI application, you should **not** block the UI thread."

-A Console app *is* A UI, and we shouldn't block that either. This is telling people to do what is almost always considered a bad practice but ok this time, so it looks dirty and it's going to confuse people.   

I ran it past Bill Wagner, who suggested two things:
Rephrase as something like "normally, we don't block the main thread, but this application doesn't allow any interaction, so it's an acceptable shortcut" I like this better as it highlights exactly why and that it is in fact a shortcut. Otherwise if they try that with a console that does need interaction, they will be hosed and it would be our guidance that told them to do it. Bill also suggested that for C#, use this syntax instead "RunAsync.GetAwaiter().GetResult();"